app/uk/gov/hmrc/gatekeeperemail/repository/FileUploadStatusRepository.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gatekeeperemail.repository
+
+import play.api.libs.json._
+
+import javax.inject.Inject
+import play.api.libs.json._
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.bson.BSONObjectID
+import reactivemongo.play.json.ImplicitBSONHandlers._
+import uk.gov.hmrc.gatekeeperemail.models.{Reference, UploadId, UploadStatus}
+import uk.gov.hmrc.gatekeeperemail.repository.UploadInfo._
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats.mongoEntity
+import uk.gov.hmrc.gatekeeperemail.models.JsonFormatters._
+import uk.gov.hmrc.gatekeeperemail.controllers.UploadDetails
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+case class UploadInfo(id : BSONObjectID, uploadId : UploadId, reference : Reference, status : UploadStatus)
+
+object UploadInfo {
+  val status = "status"
+
+
+  val format: Format[UploadInfo] = mongoEntity ( Json.format[UploadInfo] )
+}
+
+
+class FileUploadStatusRepository @Inject()(mongoComponent: ReactiveMongoComponent)(implicit ec : ExecutionContext)
```

Comment:
AFAICS we have no indexes here to support our queries. We should have indexes on `uploadId` and `reference`, I think.
Maybe consider utilising https://github.com/hmrc/reactivemongo-test/blob/master/src/main/scala/uk/gov/hmrc/mongo/FailOnUnindexedQueries.scala to help

Reply:
Furthermore, on new projects we shouldn't really be using `simple-reactivemongo`. It's effectively deprecated.
https://confluence.tools.tax.service.gov.uk/display/TEC/2021/03/04/HMRC+Mongo+is+now+available